### PR TITLE
suite-nonsplit-customers

### PR DESCRIPTION
**Pull request by himanshu-saxena-metacube**
https://spryker.atlassian.net/browse/CC-16811 - Customers

### CODE REVIEW

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/negative.robot [hunk 1]
```diff
@@ -0,0 +1,248 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/customer_endpoints/customers/negative.robot:6]: Please also add a test for customer deletion with invalid id

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    And Response body parameter should be:    [data][0][id]    ${yves_user.reference}
+    And Response body parameter should be:    [data][0][attributes][firstName]    ${yves_user.first_name}
+    And Response body parameter should be:    [data][0][attributes][lastName]    ${yves_user.last_name}
+    And Response body parameter should be:    [data][0][attributes][email]    ${yves_user.email}
+    And Response body parameter should be:    [data][0][attributes][gender]    ${gender.female}
+    And Response body parameter should be:    [data][0][attributes][salutation]    ${yves_user.salutation}
+    And Response body parameter should not be EMPTY:    [data][0][attributes][createdAt]
+    And Response body parameter should not be EMPTY:    [data][0][attributes][updatedAt]
+
+# need receive the confirmation key from email
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot:110]: Confirmation key can be received from the DB. Please use 'Save the result of a SELECT DB query to a variable:' keyword for this

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/negative.robot [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+
+Create_a_customer_with_empty_values_for_required_fields
+    I send a POST request:    /customers/    {"data":{"type":"customers","attributes":{"firstName":"","lastName":"","gender":"","salutation":"","email":"","password":"","confirmPassword":"","acceptedTerms":""}}}
+    Response status code should be:    422
+    And Response reason should be:   Unprocessable Content
+    And Response should return error code:    901
+    And Array in response should contain property with value:    [errors]    detail    firstName => This value should not be blank.
+    And Array in response should contain property with value:    [errors]    detail    lastName => This value should not be blank.
+    # And Array in response should contain property with value:    [errors]    detail    gender => This value should not be blank.
```
[the remark this answers is not in the document]
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/negative.robot]: Done

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/negative.robot (hunk 1 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/negative.robot:6]: done

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+
+ENABLER
+    TestSetup
+
+# can't receive the confirmation from email
```
[the remark this answers is not in the document]
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: Fixed

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    # ...    AND    Save value to a variable:    [data][attributes][accessToken]    token
+    # ...    AND    I set Headers:    Authorization=token
+    # ...    AND    I set Headers:    Authorization=Bearer ${token}
+    # ...    AND    Response reason should be:    OK
+    # ...    AND    Response body has correct self link internal
+    # ...    AND    I send a DELETE request:    /customers/${userId}
+    # ...    AND    Response status code should be:    204
+    # ...    AND    Response reason should be:    No Content
+
+# can't receive the confirmation from email
```
[the remark this answers is not in the document]
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: fixed

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot (hunk 2 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: fixed

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    And Response body parameter should be:    [data][attributes][email]    ${userEmail}
+    And Response body parameter should be:    [data][attributes][gender]    ${gender.male}
+    And Response body parameter should be:    [data][attributes][salutation]    ${yves_third_user.salutation}
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should not be EMPTY:    [data][attributes][createdAt]
+    And Response body parameter should not be EMPTY:    [data][attributes][updatedAt]
+    [Teardown]    Run Keywords    I send a POST request:    /access-tokens    {"data":{"type":"access-tokens","attributes":{"username":"${userEmail}","password":"${yves_user.password}"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][attributes][accessToken]    token
+    ...    AND    I set Headers:    Authorization=token
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot:35]: You don't need to set this header, it's not a required token

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    ...    AND    Response status code should be:    204
+    ...    AND    Response reason should be:    No Content
+
+
+New_customer_can_login_after_confirmation
+    [Setup]    Run Keywords    I send a POST request:    /customers/    {"data":{"type":"customers","attributes":{"firstName":"${yves_third_user.first_name}","lastName":"${yves_third_user.last_name}","gender":"${gender.male}","salutation":"${yves_third_user.salutation}","email":"${email.name}${random}${email.domain}","password":"${yves_user.password}","confirmPassword":"${yves_user.password}","acceptedTerms":True}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    userId
+    ...    AND    Save value to a variable:    [data][attributes][email]    userEmail
+    ...    AND    Save the result of a SELECT DB query to a variable:  select registration_key from spy_customer where customer_reference = '${user_id}'    confirmation_key
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot:49]: ${user_id} value that you use here doesn't equal to the userId value that you save in the row 47. Please make the variable name the same

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    I send a POST request:    /customer-confirmation   {"data":{"type":"customer-confirmation","attributes":{"registrationKey":"${confirmation_key}"}}}
+    And Response status code should be:    204
+    And Response reason should be:    No Content
+    I send a POST request:    /access-tokens    {"data":{"type":"access-tokens","attributes":{"username":"${userEmail}","password":"${yves_user.password}"}}}
+    And Response status code should be:    201
+    And Save value to a variable:    [data][attributes][accessToken]    token
+    And I set Headers:    Authorization=token
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot:56]: You don't need to set this header, it's not a required token

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot (hunk 6 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: Fixed

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot (hunk 7 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: Fixed

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot (hunk 8 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: Fixed

File: tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot (hunk 6 above)
Comment (Katerynaa-Lavryk) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: Setting of header is removed, thanks. 
Currently you create authorisation token but you do not use it in any of next steps. I guess creation of token is not required here as it is already created.
My suggestion is that you need to re-check if teardown needs to create another token and remove the steps for token creation if it's not needed there.

Comment (Katerynaa-Lavryk) [reply tests/api/suite_nonsplit/glue/customer_endpoints/customers/positive.robot]: Also you need to resolve conflicts

**Review by Katerynaa-Lavryk — APPROVED** [review]

**Pull request merged**